rust/processor/src/models/ans_models/ans_lookup.rs

```diff
@@ -0,0 +1,356 @@
+        let table_handle = standardize_address(&write_table_item.handle.to_string());
+        if table_handle
+            != standardize_address(
+                ans_v1_primary_names_table_handle
+                    .clone()
+                    .unwrap_or_default()
+                    .as_str(),
+            )
+        {
+            return Ok(None);
+        }
+        if let Some(data) = write_table_item.data.as_ref() {
+            // Return early if key is not address type. This should not be possible but just a precaution
+            // in case we input the wrong table handle
+            if data.key_type != "address" {
+                return Ok(None);
+            }
+            let decoded_key: String = serde_json::from_str(data.key.as_str()).unwrap();
+            let registered_address = standardize_address(decoded_key.as_str());
+            let value_type_name = get_name_from_unnested_move_type(data.value_type.as_ref());
+            if let Some(AnsTableItem::NameRecordKeyV1(name_record_key)) =
+                &AnsTableItem::from_table_item(value_type_name, &data.value, txn_version)?
+            {
+                return Ok(Some((
+                    Self {
+                        registered_address: registered_address.clone(),
+                        domain: Some(name_record_key.get_domain_trunc()),
+                        subdomain: Some(name_record_key.get_subdomain_trunc()),
+                        token_name: Some(name_record_key.get_token_name()),
+                        last_transaction_version: txn_version,
+                        is_deleted: false,
+                    },
+                    AnsPrimaryName {
+                        transaction_version: txn_version,
+                        write_set_change_index,
+                        registered_address,
+                        domain: Some(name_record_key.get_domain_trunc()),
+                        subdomain: Some(name_record_key.get_subdomain_trunc()),
+                        token_name: Some(name_record_key.get_token_name()),
+                        is_deleted: false,
+                    },
+                )));
+            }
+        }
+        Ok(None)
+    }
+
+    // Parse primary name from delete table item
+    // We need to lookup which domain the address points to so we can mark it as non-primary.
+    pub fn parse_primary_name_record_from_delete_table_item_v1(
+        delete_table_item: &DeleteTableItem,
+        ans_v1_primary_names_table_handle: &Option<String>,
+        txn_version: i64,
+        write_set_change_index: i64,
+    ) -> anyhow::Result<Option<(Self, AnsPrimaryName)>> {
+        let table_handle = standardize_address(&delete_table_item.handle.to_string());
+        if table_handle
+            != standardize_address(
+                ans_v1_primary_names_table_handle
+                    .clone()
+                    .unwrap_or_default()
+                    .as_str(),
```

Comment:
Did I add this? hmm if I did then prolly got lazy. 

this is weird syntax. You should limit the use of unwrap_or_default to unexpected edge cases. Better to use .map or if let Some.

Also .clone....as_str usually means that you're copying data that you don't have to.